### ISSUE TITLE
WidgetLoader fails to resolve HTML & CSS when widget-path != filename

### ISSUE THREAD

**Issue by jpommerening** [LaxarJS/laxar#225]
The problem occurs when widgets are installed at a path that does not match the widget's internal naming (for example `laxar-popup-widget` vs [`ax-popup-widget`](/LaxarJS/ax-popup-widget))

To reproduce, one can install (#129) the above widget with:
```console
$ bower install laxar-popup-widget=LaxarJS/ax-popup-widget#^2.1.0
bower laxar-popup-widget#^2.1.0 cached git://github.com/LaxarJS/ax-popup-widget.git#2.1.0
```

The `resources.json` contains the correct paths, and the `ax-popup-widget.js`-file will be loaded properly but the file resource provider will fail to provide HTML and CSS.

I'll push a "proof of concept" fix, that …
- first checks for the files as specified in the `widget.json` (with camel case converted to dashes)
- then checks for the files that match the last part of the widget path (for backwards compatibility)

**Comment by x1B** [comment]
_My 2 cts:_ For the first check, let's not apply any magic at all (like with controls): the filename should match exactly.
This does not break backwards compatibility since AMD-based installation has not been possible in the past. And in the future, it will hopefully cause less confusion. 

*Drawback:* we'll need to update all published standalone widgets. But we will need to do that anyway, because the _bower.json_ manifests currently have invalid names (containing slashes) which cause trouble when updating dependencies.

I'm filing the tickets right now :/

**Comment by x1B** [comment]
*Additional Info: * Currently, the AngularJS widget adapter relies on CamelCase `name` entries in the `widget.json`. We might need to change that.

**Comment by jpommerening** [comment]
@x1B: I don't know if that's the AngularJS widget adapter, but in addition to the controller name, there's also the name of the CSS class that's attached to the `div` that wraps the widget. This seems to depend on the `name` entry as well.

**Comment by x1B** [comment]
@jpommerening yes, that is true. Historically, the class has been normalized to *dash-separated-style*, which we probably want to ensure regardless of the actual name (to not to break existing themes if nothing else).

**Comment by x1B** [comment]
Fixed on `master` (`v1.1.0`).
